packages/pigeon/lib/dart_generator.dart

```diff
@@ -263,14 +264,26 @@ void _writeFlutterApi(
   });
 }
 
-String _addGenericTypes(String dataType, String nullTag) {
-  switch (dataType) {
+String _flattenTypeArguments(List<TypeArgument> args, String nullTag) {
```

Comment:
A comment with an example of what exactly the flattened representation is would be useful.

Reply:
Done, I added a docstring that explains it in prose.